OHIF/Viewers
# feat(filters): save worklist query filters to session storage so that they persist between navigation to the viewer and back

### Context

See #1437 and thanks go the authors and contributors of that PR.

In summary, the OHIF worklist query results filter should persist between navigation to the viewer and back to the worklist.



### Changes & Results

- Store the various filters as an object to session storage
- Because the filters are also in the URL, any refresh or changes to the URL must clear the filters from session storage so that the URL takes precedence. This does not affect navigation to/from the viewer because the React router is being used which does not trigger an unload.



### Testing

Test A: 
1. Load the OHIF worklist.
2. Add various filters to one or more of the columns
3. Open a study into the viewer using one of the worklist mode buttons.
4. Return to the worklist.
5. The filters from step 2 should be maintained.

Test B:
1. Load the OHIF worklist AND add filters to the URL.
2. The filters should be applied in the worklist GUI that displays.
3. Open a study into the viewer using one of the worklist mode buttons.
4. Return to the worklist.
5. The filters from steps 1 and 2 should be maintained.

Test C:
1. Load the OHIF worklist.
2. Add various filters to one or more of the columns
3. Notice the filters are applied and the URL is updated with the filters as query parameters
4. Edit the URL and press <Enter> to refresh/reload the page with the new URL.
5. The new URL and its filters should be applied.



### Checklist

#### PR



  semantic-release format and guidelines.

#### Code

  etc.)

#### Public Documentation Updates



  additions or removals.

#### Tested Environment

- [x] Node version: 18.16.1
- [x] Browser: Chrome 118.0.5993.89
  


[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch
